### PR TITLE
OCS-336 and OCS-346 - Test ocs basics

### PR DESCRIPTION
**Pull request by shreekarSS**
resources/pod.py
 - Added function get_admin_key_from_ceph_tools(),which returns admin keyring in base64

ocs/defaults.py
 - Added default paths for csi templates for cephfs and rbd(storageclass,secret,pvc)
tests/manage/OCS_336_346.py
 - Added two cases OCS-336 and OCS-346, which covers basics such as secret creation, storage class creation with cephfs and rbd

resources/ocs.py
 - Added return statement to delete()

### CODE REVIEW

File: resources/ocs.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def create(self):
         return status
 
     def delete(self):
-        self.ocp.delete(resource_name=self.name)
+        return self.ocp.delete(resource_name=self.name)
```
Comment (RazTamir) [inline resources/ocs.py]: Good catch!

File: resources/pod.py [hunk 2]
```diff
@@ -153,3 +154,15 @@ def get_ceph_tools_pod():
     assert ct_pod, f"No Ceph tools pod found"
     ceph_pod = Pod(**ct_pod)
     return ceph_pod
+
+
+def get_admin_key_from_ceph_tools():
+    """
+    Fetches admin key secret from ceph
+    Returns:
```
Comment (RazTamir) [inline resources/pod.py]:
```python
"""
Fetches admin key secret from ceph

Returns:
    str: Admin keyring encoded with base64 as a string
```

File: tests/manage/OCS_336_346.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        self.cephfs_secret = defaults.CEPHFS_SECRET.copy()
+        del self.cephfs_secret['data']['userID']
+        del self.cephfs_secret['data']['userKey']
+        self.cephfs_secret['data']['adminKey'] = (
+            get_admin_key_from_ceph_tools()
+        )
+        self.cephfs_secret['data']['adminID'] = defaults.ADMIN_BASE64
+        logging.info(self.cephfs_secret)
+        secret = OCS(**self.cephfs_secret)
+        assert secret.create()
```
Comment (RazTamir) [inline tests/manage/OCS_336_346.py]: All the asserts will fail on running ocp functions

File: tests/manage/OCS_336_346.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    Tearing down the environment
+    """
+    global CEPH_OBJ
+    CEPH_OBJ.delete()
+
+
+@tier1
+class TestOSCBasics(ManageTest):
+    mons = """
+        rook-ceph-mon-a.rook-ceph.svc.cluster.local:6789,
```
Comment (prsurve) [inline tests/manage/OCS_336_346.py]: replace rook-ceph with openshift-storage or use {ENV_DATA['cluster_namespace']}

File: tests/manage/OCS_336_346.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        rook-ceph-mon-b.rook-ceph.svc.cluster.local:6789,
```
Comment (prsurve) [inline tests/manage/OCS_336_346.py]: same..

File: tests/manage/OCS_336_346.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        rook-ceph-mon-c.rook-ceph.svc.cluster.local:6789
```
Comment (prsurve) [inline tests/manage/OCS_336_346.py]: same..

File: tests/manage/OCS_336_346.py (hunk 4 above)
Comment (shreekarSS) [reply tests/manage/OCS_336_346.py]: Done, Thanks @prsurve

**Comment by vasukulkarni** [comment]
Awesome to see this!

File: tests/manage/OCS_336_346.py (hunk 3 above)
Comment (shreekarSS) [reply tests/manage/OCS_336_346.py]: Removed all assert checkings

File: resources/pod.py (hunk 2 above)
Comment (shreekarSS) [reply resources/pod.py]: Done!

File: ocs/defaults.py [hunk 7]
```diff
@@ -83,3 +83,50 @@
         TEMPLATE_DIR, "PersistentVolumeClaim_new.yaml"
     )
 )
+
+CSI_TEMPLATE_DIR = os.path.join(TOP_DIR, "templates/CSI/")
+
+CEPHFS_SC = load_yaml_to_dict(
```
Comment (clacroix12) [inline ocs/defaults.py]: Is it really necessary for us to have an entry in defaults for every yaml file? This seems like overkill to me.

File: tests/manage/OCS_336_346.py [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        self.cephfs_secret['data']['adminID'] = defaults.ADMIN_BASE64
+        logging.info(self.cephfs_secret)
+        secret = OCS(**self.cephfs_secret)
+        secret.create()
+        self.cephfs_sc = defaults.CEPHFS_SC.copy()
+        self.cephfs_sc['parameters']['monitors'] = self.mons
+        self.cephfs_sc['parameters']['pool'] = (
+            f"{self.fs_data['metadata']['name']}-data0"
+        )
+        storage_class = OCS(**self.cephfs_sc)
```
Comment (shylesh) [inline tests/manage/OCS_336_346.py]: You are supposed to use StorageClass() right ? why OCS()

File: tests/manage/OCS_336_346.py [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        storage_class.create()
+        self.cephfs_pvc = defaults.CEPHFS_PVC.copy()
+        pvc = OCS(**self.cephfs_pvc)
```
Comment (shylesh) [inline tests/manage/OCS_336_346.py]: Here PVC() class right ?

File: tests/manage/OCS_336_346.py (hunk 8 above)
Comment (shylesh) [reply tests/manage/OCS_336_346.py]: Sorry ignore these comments, @RazTamir told me the reason behind single OCS class for all purpose.

File: tests/manage/OCS_336_346.py (hunk 9 above)
Comment (shylesh) [reply tests/manage/OCS_336_346.py]: Ignore.

File: ocs/defaults.py (hunk 7 above)
Comment (vasukulkarni) [reply ocs/defaults.py]: There are only couple of standard yaml's , we could do in some standard for loop with file syntax name but for now it looks fine to me to do in one place.

File: tests/manage/OCS_336_346.py [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+    """
+    Setting up the environment for the test
+    """
+    global CEPH_OBJ
+    self.fs_data = defaults.CEPHFILESYSTEM_DICT.copy()
+    self.fs_data['metadata']['name'] = helpers.create_unique_resource_name(
+        'test', 'cephfs'
+    )
+    self.fs_data['metadata']['namespace'] = ENV_DATA['cluster_namespace']
+    CEPH_OBJ = OCS(**self.fs_data)
```
Comment (ebenahar) [inline tests/manage/OCS_336_346.py]: you can replace lines 39 43 with helpers.create_resource()

File: tests/manage/OCS_336_346.py [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+        del self.rbd_secret['data']['kubernetes']
+        self.rbd_secret['data']['admin'] = get_admin_key_from_ceph_tools()
+        logging.info(self.rbd_secret)
+        secret = OCS(**self.rbd_secret)
+        secret.create()
+        self.rbd_sc = defaults.RBD_SC.copy()
+        self.rbd_sc['parameters']['monitors'] = self.mons
+        del self.rbd_sc['parameters']['userid']
+        logging.info(self.rbd_sc['parameters'])
+        storage_class = OCS(**self.rbd_sc)
```
Comment (ebenahar) [inline tests/manage/OCS_336_346.py]: maybe I'm missing something, but where are you creating the pool?

Comment (shreekarSS) [reply tests/manage/OCS_336_346.py]: I'm not creating any pool, using default pool 'rbd'.

**Comment by vasukulkarni** [comment]
@shreekarSS please post the test results, thanks!

**Comment by shreekarSS** [comment]
> @shreekarSS please post the test results, thanks!

here : http://pastebin.test.redhat.com/770542

File: resources/ocs.py (hunk 1 above)
Comment (shreekarSS) [reply resources/ocs.py]: Thanks!

**Pull request closed without merging**